controller/internal/enforcer/envoyauthorizer/envoyproxy/auth_server.go

```diff
@@ -0,0 +1,462 @@
+	}
+	zap.L().Info("ext_authz ingress: Request accepted for", zap.String("dst: ", destIP), zap.String("src: ", sourceIP))
+	zap.L().Debug("ext_authz ingress: Access authorized by network policy", zap.String("puID", s.puID))
+	return &ext_auth.CheckResponse{
+		Status: &status.Status{
+			Code: int32(code.Code_OK),
+		},
+		HttpResponse: &ext_auth.CheckResponse_OkResponse{
+			OkResponse: &ext_auth.OkHttpResponse{},
+		},
+	}, nil
+}
+
+// egressCheck implements the AuthorizationServer for egress connections
+func (s *AuthServer) egressCheck(_ context.Context, checkRequest *ext_auth.CheckRequest) (*ext_auth.CheckResponse, error) {
+	zap.L().Info("ext_authz egress: checkRequest", zap.String("puID", s.puID), zap.String("checkRequest", checkRequest.String()))
+
+	var sourceIP, destIP string
+	var source, dest *ext_auth.AttributeContext_Peer
+	var httpReq *ext_auth.AttributeContext_HttpRequest
+	var destPort, srcPort int
+	var urlStr, method string
+	attrs := checkRequest.GetAttributes()
+	if attrs != nil {
+		source = attrs.GetSource()
+		dest = attrs.GetDestination()
+
+		if source != nil {
+			if addr := source.GetAddress(); addr != nil {
+				if sockAddr := addr.GetSocketAddress(); sockAddr != nil {
+					sourceIP = sockAddr.GetAddress()
+					srcPort = int(sockAddr.GetPortValue())
+				}
+			}
+		}
+		if dest != nil {
+			if destAddr := dest.GetAddress(); destAddr != nil {
+				if destSockAddr := destAddr.GetSocketAddress(); destSockAddr != nil {
+					destIP = destSockAddr.GetAddress()
+					destPort = int(destSockAddr.GetPortValue())
+				}
+			}
+		}
+
+		if request := attrs.GetRequest(); request != nil {
+			httpReq = request.GetHttp()
+			urlStr = httpReq.GetPath()
+			method = httpReq.GetMethod()
+		}
+	}
+	// Create the new target URL based on the path parameter that we have from envoy.
+	URL, err := url.ParseRequestURI(urlStr)
+	if err != nil {
+		zap.L().Error("ext_authz egress: Cannot parse the URI", zap.Error(err))
+		return nil, err
+	}
+
+	authRequest := &apiauth.Request{
+		OriginalDestination: &net.TCPAddr{IP: net.ParseIP(destIP), Port: destPort},
+		SourceAddress:       &net.TCPAddr{IP: net.ParseIP(sourceIP), Port: srcPort},
+		URL:                 URL,
+		Method:              method,
+		RequestURI:          "",
+	}
+	r := new(http.Request)
+	r.RemoteAddr = sourceIP
+
+	resp, err := s.auth.ApplicationRequest(authRequest)
+	if err != nil {
+		if resp.PUContext != nil {
+			state := flowstats.NewAppConnectionState(s.puID, r, authRequest, resp)
+			state.Stats.Action = resp.Action
+			state.Stats.PolicyID = resp.NetworkPolicyID
+			s.collector.CollectFlowEvent(state.Stats)
+		}
+		zap.L().Error("ext_authz egress: Access *NOT* authorized by network policy", zap.String("puID", s.puID), zap.Error(err))
+		//flow.DropReason = "access not authorized by network policy"
+		return createDeniedCheckResponse(code.Code_PERMISSION_DENIED, envoy_type.StatusCode_Forbidden, "Access not authorized by network policy"), err
```

Comment:
does this mean we are not collecting flow logs in every case?

Reply:
yeah, if the PUContext is nil then we don't collect the flows.